[receiver/cloudflare] TestReceiverTLSIntegration integration test failing

### Component(s)

receiver/cloudflare

### Describe the issue you're reporting

This integration test was recently re-enabled as a part of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32532, but it continues to fail intermittently.

[Failure CI/CD link](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8778303387/job/24084569959?pr=32586)

Failure output:
```
Running target 'mod-integration-test' in module 'receiver/cloudflarereceiver' as part of group 'receiver-0'
make --no-print-directory -C receiver/cloudflarereceiver mod-integration-test
running go integration test ./... in /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/cloudflarereceiver
/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/gotestsum --rerun-fails=1 --packages="./..." -- -race -timeout 360s -parallel 4 -tags=integration,"" -run=Integration
∅  internal/metadata
✖  . (149ms)

DONE 3 tests, 2 failures in 1.127s

✖  . (83ms)

=== Failed
=== FAIL: . TestReceiverTLSIntegration/multiple_log_payload (0.06s)
    logs_integration_test.go:114: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/cloudflarereceiver/logs_integration_test.go:114
        	Error:      	Received unexpected error:
        	            	the following errors occurred:
        	            	 -  resources are out of order: resource "map[]" expected at index 0, found at index 1
        	            	 -  resources are out of order: resource "map[cloudflare.zone:example.com]" expected at index 1, found at index 2
        	            	 -  resources are out of order: resource "map[cloudflare.zone:abc.com]" expected at index 2, found at index 0
        	Test:       	TestReceiverTLSIntegration/multiple_log_payload

=== FAIL: . TestReceiverTLSIntegration (0.13s)

=== FAIL: . TestReceiverTLSIntegration/multiple_log_payload (re-run 1) (0.07s)
    logs_integration_test.go:114: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/cloudflarereceiver/logs_integration_test.go:114
        	Error:      	Received unexpected error:
        	            	the following errors occurred:
        	            	 -  resources are out of order: resource "map[]" expected at index 0, found at index 1
        	            	 -  resources are out of order: resource "map[cloudflare.zone:example.com]" expected at index 1, found at index 2
        	            	 -  resources are out of order: resource "map[cloudflare.zone:abc.com]" expected at index 2, found at index 0
        	Test:       	TestReceiverTLSIntegration/multiple_log_payload

=== FAIL: . TestReceiverTLSIntegration (re-run 1) (0.07s)

DONE 2 runs, 5 tests, 4 failures in 1.947s
make[2]: *** [../../Makefile.Common:142: mod-integration-test] Error 1
make[1]: *** [Makefile:165: receiver/cloudflarereceiver] Error 2
make: *** [Makefile:122: gointegration-test] Error 2
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
```

--- a/receiver/cloudflarereceiver/logs_integration_test.go
+++ b/receiver/cloudflarereceiver/logs_integration_test.go
@@ -111,7 +111,7 @@ func TestReceiverTLSIntegration(t *testing.T) {
 			expectedLogs, err := golden.ReadLogs(filepath.Join("testdata", "processed", fmt.Sprintf("%s.yaml", payloadName)))
 			require.NoError(t, err)
 
-			require.NoError(t, plogtest.CompareLogs(expectedLogs, logs, plogtest.IgnoreObservedTimestamp()))
+			require.NoError(t, plogtest.CompareLogs(expectedLogs, logs, plogtest.IgnoreObservedTimestamp(), plogtest.IgnoreResourceLogsOrder()))
 		})
 	}
 }

